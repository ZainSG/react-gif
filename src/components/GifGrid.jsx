import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'
import PropTypes from 'prop-types'
//import { getGifs } from '../helpers/getGifs'
const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);
    //data:images renombre la variables destructurada
  return (
      <>
    <h3 className='nimate__animated animate__flash'>{category}</h3>
    {loading && <h3>Loading...</h3>}
     <div className='card-grid'>
        {
            images.map( img => 
            <GifGridItem key={img.id} {...img}/>)
        }
       
    </div> 
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
export default GifGrid