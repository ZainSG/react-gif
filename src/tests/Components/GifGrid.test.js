import {shallow} from 'enzyme'
import '@testing-library/jest-dom'
import GifGrid from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas con el componente <GifGrid />', () => { 

    let category = 'Akame ga Kill'
    
    test('debe renderizar el componente', () => { 
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de mostrar items cuando se cargan', () => { 
        const gifs = [{
            id:'ABC',
            url: 'https://localhost/images/test.jpg',
            title:'this is title'
        },
        {
            id:'ABCD',
            url: 'https://localhost/images/test.jpg',
            title:'this is title'
        }
    ]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot()
        expect( wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)


    })


 })