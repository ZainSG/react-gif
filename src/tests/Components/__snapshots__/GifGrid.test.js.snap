// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <GifGrid /> debe de mostrar items cuando se cargan 1`] = `
<Fragment>
  <h3
    className="nimate__animated animate__flash"
  >
    Akame ga Kill
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="this is title"
      url="https://localhost/images/test.jpg"
    />
    <GifGridItem
      id="ABCD"
      key="ABCD"
      title="this is title"
      url="https://localhost/images/test.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas con el componente <GifGrid /> debe renderizar el componente 1`] = `
<Fragment>
  <h3
    className="nimate__animated animate__flash"
  >
    Akame ga Kill
  </h3>
  <h3>
    Loading...
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
